tests/frontend/unit/DpProcedureCoordinateInput.spec.js

```diff
@@ -11,13 +11,40 @@ import DpProcedureCoordinateInput from '@DpJs/components/procedure/basicSettings
 import shallowMountWithGlobalMocks from '@DpJs/VueConfigLocal'
 
 describe('DpProcedureCoordinateInput', () => {
-  it('should be named DpProcedureCoordinateInput', () => {
-    expect(DpProcedureCoordinateInput.hasOwnProperty('name')).toBe(true)
-    expect(DpProcedureCoordinateInput.name).toBe('DpProcedureCoordinateInput')
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = shallowMountWithGlobalMocks(DpProcedureCoordinateInput, {})
+  })
+
+  it('should match the snapshot', () => {
+    expect(wrapper.html()).toMatchSnapshot()
+  })
+
+  it('should check if coordinates are valid', () => {
+    wrapper.setData({ latitudeValue: '123', longitudeValue: '456' })
+    expect(wrapper.vm.isCoordinatesValid).toBe(true)
+
+    wrapper.setData({ latitudeValue: 'abc', longitudeValue: '456' })
+    expect(wrapper.vm.isCoordinatesValid).toBe(false)
   })
 
-  it('button should be enabled with valid input from props', () => {
-    const instance = shallowMountWithGlobalMocks(DpProcedureCoordinateInput, {})
-    expect(instance.html()).toEqual(expect.not.stringContaining('class="btn--disabled"'))
+  it('should update coordinates correctly', () => {
+    wrapper.vm.updateCoordinates(['789', '012'])
+    expect(wrapper.vm.latitudeValue).toBe('789')
+    expect(wrapper.vm.longitudeValue).toBe('012')
+  })
+
+  it('should convert string to float correctly', () => {
+    expect(wrapper.vm.convertToFloat('123.456')).toBe(123.456)
```

Comment:
```suggestion
   it('should convert string to float correctly', async () => {
    await wrapper.setProps({coordinate: ['123', '456']})

    expect(typeof wrapper.vm.latitudeValue).toBe('string')
    expect(typeof wrapper.vm.longitudeValue).toBe('string')
    expect(typeof wrapper.vm.convertToFloat(wrapper.vm.latitudeValue)).toBe('number')
    expect(typeof wrapper.vm.convertToFloat(wrapper.vm.longitudeValue)).toBe('number')
  })
```

I think this tests more appropriate the conversion of the values. Also writing it like that makes sure the interaction between setting props and having valid lat/long values is correct (imho)

Reply:
I think that's a valid test, but imo it does not replace the existing test, because it checks the method with a variety of inputs, including a string with a comma. I added your approach as a new test in [f9f5ca4](https://github.com/demos-europe/demosplan-core/pull/3097/commits/f9f5ca4617f14c86830b4d352931f46479e267f0)